examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -0,0 +1,320 @@
+};
+void main() {
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    array_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data[row][col];
+}
+END
+
+COMPILE_SHADER array_to_blocked_array_compiled SHADER array_to_blocked_array
+CREATE_PROGRAM array_to_blocked_array_prog SHADERS array_to_blocked_array_compiled
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM array_to_blocked_array_prog
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array1_mat_buf correct_array1_mat_buf
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM array_to_blocked_array_prog
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array2_mat_buf correct_array2_mat_buf
+
+DECLARE_SHADER mat_mul KIND COMPUTE
+#version 450
+
+#define KERNEL_SIZE 4
+#define MAT_SIZE 16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding=0) buffer array1_mat_buf{
+  mat4x4 array1_mat[MAT_SIZE/KERNEL_SIZE][MAT_SIZE/KERNEL_SIZE];
+};
+layout(std430, binding=1) buffer array2_mat_buf{
+  mat4x4 array2_mat[MAT_SIZE/KERNEL_SIZE][MAT_SIZE/KERNEL_SIZE];
+};
+layout(std430, binding=2) buffer blocked_result{
+  mat4x4 blocked_res[MAT_SIZE/KERNEL_SIZE][MAT_SIZE/KERNEL_SIZE];
+};
+
+void main() {
+    for(int i = 0;i < MAT_SIZE/KERNEL_SIZE; i++){
+        for(int j = 0; j < MAT_SIZE/KERNEL_SIZE; j++){
+            blocked_res[i][j] = mat4x4(0);
+            for(int k = 0; k < MAT_SIZE/KERNEL_SIZE; k++){
+                 // The submatrix at position (i,j) is equal to the multiplication
+                // of all the submatrices at (i,k) and (k,j) where k = 0....(MAT_SIZE/KERNEL_SIZE).
+                blocked_res[i][j] += array1_mat[i][k] * array2_mat[k][j];
+            }
+        }
+    }
+}
+END
+
+COMPILE_SHADER mat_mul_compiled SHADER mat_mul
+CREATE_PROGRAM mat_mul_prog SHADERS mat_mul_compiled
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+BIND_SHADER_STORAGE_BUFFER BUFFER blocked_result BINDING 2
+
+RUN_COMPUTE
+    PROGRAM mat_mul_prog
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS blocked_result correct_blocked_result
+
+DECLARE_SHADER blocked_array_to_array KIND COMPUTE
+#version 450
+
+#define KERNEL_SIZE 4
+#define MAT_SIZE 16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding=0) buffer result{
+  float res[MAT_SIZE][MAT_SIZE];
+};
+layout(std430, binding=1) buffer blocked_result{
+  mat4x4 blocked_res[MAT_SIZE/KERNEL_SIZE][MAT_SIZE/KERNEL_SIZE];
+};
+
+void main(){
+  for(int i = 0; i < MAT_SIZE; i++){
+    for(int j = 0; j < MAT_SIZE; j++){
+        // Converts the 4x4 array of 4x4 matrices to a 16x16 array
+        res[i][j] = blocked_res[i/KERNEL_SIZE][j/KERNEL_SIZE][j%KERNEL_SIZE][i%KERNEL_SIZE];
+    }
+  }
+}
```

Comment:
When reviewing your other PR I realised that this compute shader isn't quite right: every workgroup unblocks the whole matrix.

Can you revise so that each workgroup takes care of unblocking a particular submatrix, so that the unblockings are disjoint and happen in parallel?